taichi/ir/analysis.h

```diff
@@ -42,7 +42,7 @@ class DiffRange {
   }
 
   bool linear_related() const {
-    return related && coeff == 1;
+    return related && coeff > 0;
```

Comment:
The term `linear_related()` is confusing here in that it's actually used to check whether `related && coeff == 1` in the codebase. Actually, as long as `related` is true, the relation is linear... So is there a better name for the function `related && coeff == 1`?

Reply:
Hmm, Do we actually need to check if coeff == 1 though? All tests pass at the moment with coeff > 0.

Reply:
I think we don't have any tests such that `related && coeff != 1` now... Maybe we should not change the logic of other places using this function in this PR.

Reply:
So the only reason I needed related && coeff > 1 was to make it into this if statement. I could make the it just check if diff.related_? But I'm not sure if thats coeff = 0 would be well behaved. Should I just make a separate function for coeff > 0? Trying to think of a a good name for it.  

https://github.com/taichi-dev/taichi/blob/df0ee9e7861abeec1322b9a7411991798fa56596/taichi/transforms/insert_scratch_pad.cpp#L90-L98

Reply:
I went ahead and reverted linear related back to related & coeff == 1. changed the if statment there to check diff.related_() && diff.coeff > 0